lib/rrd_utils.ml

```diff
@@ -20,6 +20,29 @@
 
 exception Parse_error
 
+module BoundedFloat : sig
+  type t = private float
+  type entry_fun = Clamp | To_Nan
+  val of_float: minimum:float -> maximum:float -> f:entry_fun -> float -> t
+  val to_float: t -> float
+end = struct
+  type t = float
+  type entry_fun = Clamp | To_Nan
+
+  let of_float ~minimum ~maximum ~f x = match f with
+  | Clamp -> min maximum (max minimum x)
+  | To_Nan when x < minimum || x > maximum -> nan
+  | To_Nan -> x
+
+  let to_float x = x
+end
+
+(* cast double-precision float to single-precision *)
+let castd2s f =
```

Comment:
Can you do the calls to this only from unit-tests, and move this to the unit test code?
This allocates a new bigarray every time, which is more costly than allocating an OCaml value.

Reply:
Sure, I can be part of the tests only.

I didn't think it would add that much overhead, datasources are created at very specific points in the daemon/plugin lifetimes. (startup and deserializing archives from disk)

Reply:
Function is now only used in crowbar tests.